File: java-implementation/src/main/java/com/inventory/transport/Air.java
package com.inventory.transport;

import com.inventory.transport.TransportTypes.AirType;
import static com.inventory.transport.TransportTypes.AirTransportProperties;
import static com.inventory.transport.TransportTypes.AirRoutes;

// Represent the properties of air transport
public class Air extends Transport {
    private AirType type;
    private String airRoutes;

    // Constructs an Air transport instance with properties based on the properties
    public Air(AirType type) {

         // Call the base class constructor with speed and cost
        super(AirTransportProperties.get(type).getFirst(), AirTransportProperties.get(type).getSecond());
        this.type = type;
        this.airRoutes = AirRoutes.get(type);
    }

    // Calculates delivery time
    @Override
    public double calculateDeliveryTime(double distance) {
        return distance / speed;
    }

    // Calculates delivery cost
    @Override
    public double calculateDeliveryCost(double distance) {
        return distance * costPerKm;
    }

    // Retrieves the transport type
    @Override
    public String getTransportType() {
        return toString(type);
    }

    // Returns the predetermined stages of a route based on the transport type
    @Override
    public String getRouteStages() {
        return airRoutes;
    }

    // Converts the enum to a legible string
    private String toString(AirType type) {
        switch (type) {
            case CargoPlane: return "CargoPlane";
            case CommercialPlane: return "CommercialPlane";
            case Helicopter: return "Helicopter";
            default: return "Unknown";
        }
    }
}

File: cpp-implementation/src/transport/Air.cpp
#include "Transport.ih"

// Constructor for the Air class
Air::Air(AirType type)
    : Transport(AirTransportProperties.at(type).first, AirTransportProperties.at(type).second),
      type(type), airRoutes(AirRoutes.at(type)) {
    
}

// Calculates the time taken to cover a distance at the object's speed.
double Air::calculateDeliveryTime() const {
 
    return distance / speed;
}

// Calculates the total cost of delivering over a distance.
double Air::calculateDeliveryCost() const {
    return distance * costPerKm;
}

// Converts the OceanType enum value to a readable string.
string Air::getTransportType() const {
    return toString(type);
}

// Returns a string describing the route stages for the specific ocean transport.
string Air::getRouteStages() const {
    return airRoutes;
}

// Helper function to convert OceanType enum to string
string toString(AirType type) {
    switch (type) {
        case AirType::CargoPlane: return "CargoPlane";
        case AirType::CommercialPlane: return "CommercialPlane";
        case AirType::Helicopter: return "Helicopter";
        default: return "Unknown";
    }
}
